scripts/metric_reporter/reporter/coverage_reporter.py

```diff
@@ -0,0 +1,126 @@
+            "Line Covered": self.line_covered,
+            "Line Not Covered": self.line_not_covered,
+            "Line Excluded": self.line_excluded,
+            "Line Percent": self.line_percent,
+            "Function Count": self.function_count,
+            "Function Covered": self.function_covered,
+            "Function Not Covered": self.function_not_covered,
+            "Function Percent": self.function_percent,
+            "Branch Count": self.branch_count,
+            "Branch Covered": self.branch_covered,
+            "Branch Not Covered": self.branch_not_covered,
+            "Branch Percent": self.branch_percent,
+        }
+
+
+class CoverageReporter(BaseReporter):
+    """Handles the reporting of coverage results."""
+
+    def __init__(
+        self,
+        repository: str,
+        workflow: str,
+        test_suite: str,
+        coverage_artifact_list: list[CoverageSummary] | None,
+    ) -> None:
+        """Initialize the reporter with the coverage data.
+
+        Args:
+            repository (str): The repository associated to the test suite.
+            workflow (str): The workflow associated to the test suite.
+            test_suite (str): The test suite name.
+            coverage_artifact_list (list[CoverageSummary]): The coverage summary data from test
+                                                            suites.
+        """
+        super().__init__()
+        self.results: Sequence[CoverageReporterResult] = self._parse_results(
+            repository, workflow, test_suite, coverage_artifact_list
+        )
+
+    def _parse_results(
+        self,
+        repository: str,
+        workflow: str,
+        test_suite: str,
+        coverage_artifact_list: list[CoverageSummary] | None,
+    ) -> Sequence[CoverageReporterResult]:
+        if coverage_artifact_list is None:
+            return []
+
+        return [
+            CoverageReporterResult(
+                repository=repository,
+                workflow=workflow,
+                test_suite=test_suite,
+                timestamp=artifact.timestamp,
+                date=self._extract_date(artifact.timestamp) if artifact.timestamp else None,
+                job=artifact.job,
+                line_count=artifact.line_count,
+                line_covered=artifact.line_covered,
+                line_not_covered=artifact.line_not_covered,
+                line_excluded=artifact.line_excluded,
+                line_percent=artifact.line_percent,
+                function_count=artifact.function_count,
+                function_covered=artifact.function_covered,
+                function_not_covered=artifact.function_not_covered,
+                function_percent=artifact.function_percent,
+                branch_count=artifact.branch_count,
+                branch_covered=artifact.branch_covered,
+                branch_not_covered=artifact.branch_not_covered,
+                branch_percent=artifact.branch_percent,
```

Comment:
Not sure if it's any easier just to pass the entire `artifact` object vs 16 separate args. (but admittedly have no strong feelings one way or the other)

Reply:
Originally I had this! but the flat structure is way more comprehensible... I'm thinking to move the logic that creates CoverageSummary out of the parser and into the reporter...

Reply:
yeah seems better, https://github.com/mozilla/ecosystem-test-scripts/pull/21/commits/06421655466af51f98f4f7ebc2d634fd56d00bce, what do you think?